client/network/src/protocol/generic_proto/handler/group.rs

```diff
@@ -706,6 +706,16 @@ impl ProtocolsHandler for NotifsHandler {
 					}
 				}
 			}
+
+		} else {
+			// If `self.pending_legacy_handshake` is `None`, we don't want to accept events from
```

Comment:
```suggestion
			// If `self.pending_legacy_handshake` is `Some`, we don't want to accept events from
```

We are reaching this point only if `self.pending_legacy_handshake` is `Some`, right?